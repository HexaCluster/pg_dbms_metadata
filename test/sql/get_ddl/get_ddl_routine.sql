SELECT dbms_metadata.get_ddl('FUNCTION','merge_objects','gdmmm');

SELECT dbms_metadata.get_ddl('PROCEDURE','join_entity','gdmmm');

-- tests for schema as NULL
SET search_path TO public, gdmmm;

SELECT dbms_metadata.get_ddl('FUNCTION','merge_objects');

SELECT dbms_metadata.get_ddl('PROCEDURE','join_entity');