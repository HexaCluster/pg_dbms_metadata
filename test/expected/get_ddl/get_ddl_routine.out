                                           get_ddl                                           
---------------------------------------------------------------------------------------------
 CREATE OR REPLACE FUNCTION gdmmm.merge_objects(first_val text, actual_finder gdmmm.address)+
  RETURNS text                                                                              +
  LANGUAGE plpgsql                                                                          +
 AS $function$                                                                              +
 BEGIN                                                                                      +
     RETURN first_val || actual_finder.city;                                                +
 END                                                                                        +
 $function$                                                                                 +
 
(1 row)

                                                                   get_ddl                                                                    
----------------------------------------------------------------------------------------------------------------------------------------------
 CREATE OR REPLACE PROCEDURE gdmmm.join_entity(INOUT var_str text, INOUT num_low bigint, INOUT var_init text, INOUT mid_finder gdmmm.address)+
  LANGUAGE plpgsql                                                                                                                           +
 AS $procedure$                                                                                                                              +
 BEGIN                                                                                                                                       +
     num_low := mid_finder.pin;                                                                                                              +
     var_init := var_str || ', ' || mid_finder.city;                                                                                         +
 END                                                                                                                                         +
 $procedure$                                                                                                                                 +
 
(1 row)

SET
                                        get_ddl                                        
---------------------------------------------------------------------------------------
 CREATE OR REPLACE FUNCTION gdmmm.merge_objects(first_val text, actual_finder address)+
  RETURNS text                                                                        +
  LANGUAGE plpgsql                                                                    +
 AS $function$                                                                        +
 BEGIN                                                                                +
     RETURN first_val || actual_finder.city;                                          +
 END                                                                                  +
 $function$                                                                           +
 
(1 row)

                                                                get_ddl                                                                 
----------------------------------------------------------------------------------------------------------------------------------------
 CREATE OR REPLACE PROCEDURE gdmmm.join_entity(INOUT var_str text, INOUT num_low bigint, INOUT var_init text, INOUT mid_finder address)+
  LANGUAGE plpgsql                                                                                                                     +
 AS $procedure$                                                                                                                        +
 BEGIN                                                                                                                                 +
     num_low := mid_finder.pin;                                                                                                        +
     var_init := var_str || ', ' || mid_finder.city;                                                                                   +
 END                                                                                                                                   +
 $procedure$                                                                                                                           +
 
(1 row)

